File: php/features.php
<?php
declare(strict_types=1);

// Republicmag SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RepublicmagFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RepublicmagBaseFeature();
            case "test":
                return new RepublicmagTestFeature();
            default:
                return new RepublicmagBaseFeature();
        }
    }
}

File: py/features.py
# Republicmag SDK feature factory

from feature.base_feature import RepublicmagBaseFeature
from feature.test_feature import RepublicmagTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RepublicmagBaseFeature(),
        "test": lambda: RepublicmagTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
